scripts/install_plugin_modules.js

```diff
@@ -144,59 +141,52 @@ async function assertPackage (name, version, dependencyVersionRange, external) {
 }
 
 /**
- * @param {object} dependencies
- * @param {string} externalName
- * @param {string} versionRange
+ * @param {object} rootFolder
+ * @param {string} parent
  */
-async function addDependencies (dependencies, externalName, versionRange) {
-  const versionList = await getVersionList(externalName)
-  const version = semver.maxSatisfying(versionList, versionRange)
-  const pkgJson = await npmView(`${externalName}@${version}`)
-  for (const { dep, name } of externalDeps.get(externalName)) {
-    // do stuff with dep
-    for (const section of ['devDependencies', 'peerDependencies']) {
-      if (pkgJson[section] && name in pkgJson[section]) {
-        if (dep === externalName) {
-          dependencies[name] = version
-        } else if (pkgJson[section][name].includes('||')) {
-          // Use the first version in the list (as npm does by default)
-          dependencies[name] = pkgJson[section][name].split('||')[0].trim()
-        } else {
-          // Only one version available so use that.
-          dependencies[name] = pkgJson[section][name]
-        }
-        break
-      }
+async function assertPeerDependencies (rootFolder, parent) {
+  const entries = await readdir(rootFolder)
+
+  for (const entry of entries) {
+    const folder = join(rootFolder, entry)
+
+    if (!(await lstat(folder)).isDirectory()) continue
+    if (entry === 'node_modules') continue
+    if (entry.startsWith('@')) {
+      await assertPeerDependencies(folder, entry)
+      continue
     }
-  }
-}
 
-/**
- * @param {string} name
- * @returns {Promise<string[]>}
- */
-async function getVersionList (name) {
-  let list = versionListCache.get(name)
-  if (list) return list
-  list = await npmView(`${name} versions`)
-  versionListCache.set(name, list)
-  return list
-}
+    const externalName = [parent, entry.split('@')[0]].filter(v => v).join('/')
 
-/**
- * @param {string} input
- * @param {boolean} [retry=true]
- * @returns {Promise<any>}
- */
-function npmView (input, retry = true) {
-  return new Promise((resolve, reject) => {
-    childProcess.exec(`npm view ${input} --json`, (err, stdout) => {
-      if (err) {
-        return retry ? npmView(input, false).then(resolve, reject) : reject(err)
+    if (!externalDeps.has(externalName)) continue
+
+    const versionPkgJsonPath = join(folder, 'package.json')
+    const versionPkgJson = require(versionPkgJsonPath)
+
+    for (const { dep, name } of externalDeps.get(externalName)) {
+      const pkgJsonPath = require(folder).pkgJsonPath()
+      const pkgJson = require(pkgJsonPath)
+
+      for (const section of ['devDependencies', 'peerDependencies']) {
+        if (pkgJson[section] && name in pkgJson[section]) {
```

Comment:
Nit: It seems odd that you're mixing two different ways to detect if a property exists in an object, I'd personally just do:

```suggestion
        if (pkgJson[section]?.[name]) {
```